### PR TITLE
fix(ScyllaYaml): remove defaults

### PR DESCRIPTION
**Pull request by soyacz**
Having defaults set, it is not possible to put some value into
`scylla.yaml` because default values are skipped. This causes problems
when having default `scylla.yaml` with some value that is not default in
`config.cc` (e.g. `consistent_cluster_management`).

Remove default values from `ScyllaYaml` class, so in case we explicitly
set one, it will be always put into `scylla.yaml`.

--

When publishing `ScyllaYamlUpdateEvent` message is not shown in events
log.

Add message info to `ScyllaYamlUpdateEvent` for better visibility of
scylla.yaml changes in events log.

fixes: #5912

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by fruch — APPROVED** [review]
LGTM

**Comment by fruch** [comment]
@soyacz 

you need to handle also `unit_tests/test_scylla_yaml_builders.py` seems like they are failing cause of this change as well, even I don't expected them to fail

File: unit_tests/test_data/test_scylla_yaml_builders/PR-provision-test.result.json [hunk 1]
```diff
@@ -14,5 +14,9 @@
     }
   ],
   "experimental": true,
-  "prometheus_address": "0.0.0.0"
+  "prometheus_address": "0.0.0.0",
+  "alternator_enforce_authorization": false,
+  "auto_bootstrap": true,
+  "enable_ipv6_dns_lookup": false,
+  "hinted_handoff_enabled": "enabled"
```
Comment (fruch) [inline unit_tests/test_data/test_scylla_yaml_builders/PR-provision-test.result.json:21]: @soyacz why do we have those 4 extra ones ? where they are coming from ? the original yaml had them and then were equal to the defaults ?

**Review by fruch — DISMISSED** [review]
LGTM

**Comment by roydahan** [comment]
Let's make sure we give it a quick test.

File: unit_tests/test_data/test_scylla_yaml_builders/PR-provision-test.result.json (hunk 1 above)
Comment (soyacz) [reply unit_tests/test_data/test_scylla_yaml_builders/PR-provision-test.result.json:21]: I think it comes from `ScyllaYamlClusterAttrBuilder`, previously they were defaults -> skipped from yaml.

**Comment by soyacz** [comment]
I had to do some tuning to `hinted_handoff_enabled` as `nodetool scrub` was complaining about malformed scylla.yaml, apparently it does not accept `enabled` as value (I checked tests, and looks that we use `True` or `False` when we play with this setting in tests anyway).

Now all test passes, except Azure provision test - fails on syntax error for query `TRUNCATE ks_truncate_large_partition.test_table USING TIMEOUT 600s`.

**Comment by fruch** [comment]
@roydahan, @enaydanov, can you review ?

**Review by roydahan — APPROVED** [review]

**Comment by fgelcer** [comment]
> TRUNCATE ks_truncate_large_partition.test_table USING TIMEOUT 600

@soyacz , it probably fails because `TRUNCATE` with parameter `USING TIMEOUT` is support from 5.2 (#5965 )
i don't know which version you are using here in the test that failed, nor why you have `USING TIMEOUT` in the `TRUNCATE` command, but it is unrelated to your changes

**Comment by fgelcer** [comment]
the test is running with Scylla `5.0.5`
so where ever this this truncate command is coming from, it needs to be versioned

**Comment by roydahan** [comment]
I backported it and several other commits that it was dependent on it.
Forgot to label.
thx.

**Pull request merged**